### ISSUE TITLE
Flat Temperature Profile slope

### ISSUE THREAD

**Issue by vincep5** [beestat/app#348]
My heating temperature profile looks pretty flat across the various temperatures in the season.  I think it's related to how Ecobee reports the ending temperature.  
![image](https://user-images.githubusercontent.com/26965876/148291617-b07eca4b-f7a4-45b9-b62d-f1682820271e.png)

Example if I look at when the thermostat setpoint is 71.  My heating cycle starts at around 70.5.  The end of the heat cycle shows it finished at 70.6.  But really this is not true, the heat would continue until it reaches closer to 71.  The subsequent 5 minute interval shows its 71.0

There's quite a bit delta difference if in the 15minute cycle it is calculated from 70.5->70.8 vs 70.5->71.0

Downloading a day's data from Ecobee IQ also shows this occurrence where the next interval is more indicative of the real ending temperature reading.

**Comment by ziebelje** [comment]
Your recent data pretty consistently shows that as soon as your heat shuts off there is an increase in the indoor temperature of 0.3 - 0.5°F in the next 5 minutes. If I extended the capture time by five minutes, you would see your heat profile shift up slightly as that last temperature rise is at a faster pace than the rest of the heating cycle.

![image](https://user-images.githubusercontent.com/3288812/148660938-7a17ce6f-c4d0-48fe-ad81-f7bcaf710642.png)

It seems pretty clear that ecobee is turning your system off before it reaches your setpoint. Mine does it too, although it just shuts off 0.1°F early and I have much longer cycle times so it doesn't really affect anything.

I'm not sure what the correct solution is here. Today I just take the temperature change during the period the heat was running divided by the time the heat was running. What you're asking is to instead use the temperature change during the period the heat was running plus any latent temperature change.

That would be a pretty significant change, and outside of radiant heat systems I'm not sure why the temperature continues to increase so much for you. Have you measured the same change in temperature with a thermometer or sensor in a neutral location like the center of a room?

As for the flat profile - I would expect that for a gas system; they aren't affected by outdoor temperatures like heat pumps are.

**Comment by vincep5** [comment]
Yeah, my furnace does have this automatic feature to shutoff the gas from firing, but keep the blower running until a internal temp sensor gets cool enough.  So, it automatically dissipates some heat after the official "run" is over.  Note this is not a Ecobee setting (that is in addition if we wanted to configure it)
I do notice that having a 0.5 degree temp threshold vs 1.0 degree temp threshold probably plays into it as well.  The cycle is a bit shorter and thus probably not getting a good reading on the temp rise.  It's surely more comfortable in the house with only a 0.5 degree threshold.  And it's not making much difference in the number of cycles.  I counted just 2 more cycles during the night since I tested the change.

![image](https://user-images.githubusercontent.com/26965876/148822796-a94ff255-0c0f-46c2-b05b-66e61a64c470.png)
